File: EjercicioFuncional/frontend/Frontend/src/services/book-service.js
import axios from "axios";

const API_URL = "http://localhost:5203/api/Book/";

export async function getBooks() {
    const response = await axios.get(API_URL);
    return response.data;
}

export async function getBookById(id) {
    const response = await axios.get(`${API_URL}${id}`);
    return response.data;
}

export async function createBook(book) {
    const response = await axios.post(API_URL, book);
    return response.data;
}

export async function updateBook(id, book) {
    const response = await axios.put(`${API_URL}${id}`, book);
    return response.data;
}

export async function deleteBook(id) {
    await axios.delete(`${API_URL}${id}`);
}

File: EjercicioFuncional/backend/EstructuraDatos/Services/BookService.cs
using Entities;
using EstructuraDatos.Repositories;
using Services.Interfaces;

namespace EstructuraDatos.Services;

public class BookService : IBookService
{
    private BookRepository _bookRepository;
    private AuthorRepository _authorRepository;
    
    public BookService()
    {
        _bookRepository = new BookRepository(_authorRepository = new AuthorRepository());

    }
    public Task<List<Book>> GetAllBooks()
    {
        return _bookRepository.GetAll();
    }

    public Task<Book?> GetBookById(Guid id)
    {
        return _bookRepository.GetById(id);
    }

    public Task<Book> CreateBook(Book book)
    {
        return _bookRepository.Create(book);
    }

    public Task<bool> DeleteBook(Guid id)
    {
        return _bookRepository.Delete(id);
    }

    public Task<bool> UpdateBook(Book book)
    {
        return _bookRepository.Update(book);
    }
}